app/javascript/controllers/tally_controller.js

```diff
@@ -0,0 +1,49 @@
+import { Controller } from "@hotwired/stimulus";
+
+export default class extends Controller {
+  async showPopup() {
+    if (this.#hasAlreadyAnswered) return
+
+    await this.#loadScript()
+    this.#displayForm()
+  }
+
+  #loadScript() {
+    return new Promise((resolve) => {
+      if (!document.querySelector("script#tally-script")) {
+        const script = document.createElement("script");
+        script.src = "https://tally.so/widgets/embed.js";
+        script.id = "tally-script"
+        script.onload = resolve
+        document.head.appendChild(script);
+      } else {
+        resolve();
+      }
+    })
+  }
+
+  #displayForm() {
+    setTimeout(() => {
+      window.Tally.openPopup(this.element.dataset.tallyFormId, {
+        onSubmit: () => {
+          this.#hasAlreadyAnswered = true
+
+          // Give time for the user to read success message in Tally's popup
+          setTimeout(() => window.Tally.closePopup(this.element.dataset.tallyFormId), 1000);
+        }
+      })
+    }, this.element.dataset.tallyDelayInMs || 0)
+  }
+
+  set #hasAlreadyAnswered(value) {
```

Comment:
Il me semblait qu'on avait dit qu'on évitait les getter/setter dans le js. On peut peut-être juste renommer cette méthode (par ex `setHasAlreadyAnswered`)

Reply:
Ah bon ? Pourquoi ?

Reply:
Ah oui mais c'était différent : on faisait juste ça : 
```
  get id() {
    return this._id;
  }

  set id(id) {
    this._id = id;
  }
```
Les getters setters n'avaient aucune plus value par rapport à juste faire `this.id = X`

Dans le cas présent y'a une logique associée, le faire de faire `this.hasAlreadyAnswered = true` vient en fait set dans le local storage donc utiliser l'interface de `get` `set` a une vraie plus value ici puisqu'on créé une abstraction sur le localStorage . 

Je serais pour le laisser comme ça ici si c'est ok pour toi :)